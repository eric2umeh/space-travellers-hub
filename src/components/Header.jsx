import React from 'react';
import { Link } from 'react-router-dom';
import Navigation from './Navigation';

const Header = () => (
  <header className="header">
    <div className="header-container row">
      <Link to="/" className="logo">
        Bookstore CMS
      </Link>
      <Navigation />
      <span className="user">
        <BsFillPersonFill className="icon" />
      </span>
    </div>
  </header>
);

export default Header;
