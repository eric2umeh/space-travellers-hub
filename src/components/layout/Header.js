import { Link, NavLink } from 'react-router-dom';

import logo from '../../assets/logo.png';
import classes from './Header.module.css';

const Header = () => {
  const activeStyle = {
    textDecoration: 'underline',
  };

  return (
    <header className={classes.header}>
      <nav className={classes.nav}>
        <div className={classes.brand}>
          <div className={classes.logo_img}>
            <img src={logo} alt="Space Travlers" />
          </div>
          <div className={classes.logo_txt}>
            <Link to="/">
              <h1>Space Traveler&apos;s Hub</h1>
            </Link>
          </div>
        </div>
        <ul className={classes.nav_items}>
          <li>
            <NavLink
              to="/"
              style={({ isActive }) => (isActive ? activeStyle : undefined)}
            >
              Rockets
            </NavLink>
          </li>
          <li>
            <NavLink
              to="/missions"
              style={({ isActive }) => (isActive ? activeStyle : undefined)}
            >
              Missions
            </NavLink>
          </li>
          <li>
            <NavLink
              to="/profile"
              style={({ isActive }) => (isActive ? activeStyle : undefined)}
            >
              My Profile
            </NavLink>
          </li>
        </ul>
      </nav>
    </header>
  );
};

export default Header;
