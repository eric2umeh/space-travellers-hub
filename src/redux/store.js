import { configureStore } from '@reduxjs/toolkit';
import rockets from './rockets/rocketsSlice';

const store = configureStore({
  reducer: {
    rockets: rockets.reducer,
  },
});

export default store;
