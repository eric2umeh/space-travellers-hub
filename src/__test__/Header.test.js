import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Header from '../components/layout/Header';
import store from '../redux/store';

const App = () => {
  <Provider store={store}>
    <Header />
  </Provider>;
};

describe('Header component test', () => {
  it('Testing Header rendering', () => {
    const HeaderComponent = renderer.create(<App />).toJSON();
    expect(HeaderComponent).toMatchSnapshot();
  });
});
