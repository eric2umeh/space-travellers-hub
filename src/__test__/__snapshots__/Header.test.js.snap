// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test Testing Header rendering 1`] = `null`;
